src/main/scala/petclinic/services/PetService.scala

```diff
@@ -1,3 +1,100 @@
 package petclinic.services
 
-trait PetService {}
+import petclinic.QuillContext
+import petclinic.models.{OwnerId, Pet, PetId, Species}
+import zio._
+
+import javax.sql.DataSource
+
+trait PetService {
+
+  def create(name: String, birthdate: java.time.LocalDate, species: Species, ownerId: OwnerId): Task[Pet]
+  def delete(id: PetId): Task[Unit]
+  def get(id: PetId): Task[Option[Pet]]
+  def getAll: Task[List[Pet]]
+  def update(
+      id: PetId,
+      name: Option[String],
+      birthdate: Option[java.time.LocalDate],
+      species: Option[Species],
+      ownerId: Option[OwnerId]
+  ): Task[Unit]
+
+}
+
+object PetService {
+
+  def create(
+      name: String,
+      birthdate: java.time.LocalDate,
+      species: Species,
+      ownerId: OwnerId
+  ): ZIO[PetService, Throwable, Pet] =
+    ZIO.serviceWithZIO[PetService](_.create(name, birthdate, species, ownerId))
+
+  def delete(id: PetId): ZIO[PetService, Throwable, Unit] =
+    ZIO.serviceWithZIO[PetService](_.delete(id))
+
+  def get(id: PetId): ZIO[PetService, Throwable, Option[Pet]] =
+    ZIO.serviceWithZIO[PetService](_.get(id))
+
+  def getAll: ZIO[PetService, Throwable, List[Pet]] =
+    ZIO.serviceWithZIO[PetService](_.getAll)
+
+  def update(
+      id: PetId,
+      name: Option[String],
+      birthdate: Option[java.time.LocalDate],
+      species: Option[Species],
+      ownerId: Option[OwnerId]
+  ): ZIO[PetService, Throwable, Unit] =
+    ZIO.serviceWithZIO[PetService](_.update(id, name, birthdate, species, ownerId))
+
+}
+
+final case class PetServiceLive(random: Random, dataSource: DataSource) extends PetService {
```

Comment:
I think you probably don't need the `Random` service as a parameter here since you can just call operators that use `Random` without adding it to the environment.